src/list.ts

```diff
@@ -0,0 +1,259 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// SPDX-FileCopyrightText: 2025 Tim Beckmann <beckmann.tim@fh-swf.de>
+
+import {
+    Client,
+    Databases,
+    ID,
+    Permission,
+    RealtimeResponseEvent,
+    Role
+} from 'appwrite';
+import { CONFIG } from './config.ts';
+import type {
+    Einkaufsliste,
+    Listeneintrag
+} from './types.ts';
+
+const client: Client = new Client();
+
+client
+    .setEndpoint(CONFIG.BACKEND_ENDPOINT)
+    .setProject(CONFIG.PROJECT_ID);
+
+const database: Databases = new Databases(client);
+
+const urlParams = new URLSearchParams(window.location.search);
+
+const listid = urlParams.get('id') || '';
+
+if (listid == '') {
+    window.location.pathname = '/uebersicht.html';
+}
+
+const form = document.getElementById('newProduct') as HTMLFormElement;
+const deadline = document.getElementById('deadline') as HTMLInputElement;
+const quantity = document.getElementById('quantity') as HTMLInputElement;
+const listTitel = document.getElementById('listTitel') as HTMLInputElement;
+const productName = document.getElementById('productName') as HTMLInputElement;
+const productTable = document.getElementById('productTable') as HTMLTableElement;
+const listDescription = document.getElementById('listDescription') as HTMLTextAreaElement;
+const editDescriptionButton = document.getElementById('editDescriptionButton') as HTMLButtonElement;
+
+const shoppinglist = database.getDocument(
+    CONFIG.DATABASE_ID,
+    CONFIG.DB_COLLECTION_SHOPPINGLISTS,
+    listid
+) as Promise<Einkaufsliste>;
+
+shoppinglist.then((list) => {
+    realtimeUpdate(list);
+
+    updateListInfo(list);
+
+    editDescriptionButton.addEventListener('click', () => {
+        if (!listDescription.disabled.valueOf()) {
+            database.updateDocument(
+                CONFIG.DATABASE_ID,
+                CONFIG.DB_COLLECTION_SHOPPINGLISTS,
+                list.$id!,
+                {
+                    stichtag: deadline.value,
+                    beschriftung: listTitel.value,
+                    beschreibung: listDescription.value
+                }
+            );
+        }
+        listTitel.disabled = !listTitel.disabled.valueOf();
+        listDescription.disabled = !listDescription.disabled.valueOf();
+        deadline.disabled = !deadline.disabled.valueOf();
+    });
+
+    form.addEventListener('submit', (event) => {
+        event.preventDefault();
+
+        const product: Listeneintrag = {
+            artikelname: productName.value,
+            anzahl: Number.parseInt(quantity.value),
+            erledigt: false,
+            einkaufslisten: list.$id!
+        }
```

Comment:
Consider using a custom or derived type for this instance. See my types.ts comment.